docs/source/containers/singularity.md

```diff
@@ -0,0 +1,101 @@
+(singularity)=
+# Singularity on Discovery
+
+Singularity is the container runtime engine for the Discovery cluster and is installed as a module. With Singularity, you can run existing containers or build and run your own custom container.
+
+## How to run a container with Singularity
+
+You can run a container with the `run` or `exec` commands after moving to a compute node and loading the singularity module.
+
+To see available singularity versions run: module avail singularity
+
+### The `--bind` or `-B` flag
+
+Its important to mount the directories in discovery to the container so you can access necessary input data and to output the results of your software to the directory you specify. 
+
+### Example using a container file on the file system
+
+Here’s an `srun` example using a container file from the /shared/container_repository
+
+```bash
+srun --pty /bin/bash
+module load singularity/3.10.3
+singularity run -B "/work:/work" /shared/container_repository/star/star_2.7.8a.sif
+
+Singularity>
+
+Singularity> STAR
+Usage: STAR  [options]... --genomeDir /path/to/genome/index/   --readFilesIn R1.fq R2.fq
+Spliced Transcripts Alignment to a Reference (c) Alexander Dobin, 2009-2020
+
+STAR version=2.7.8a
+STAR compilation time,server,dir=Wed Nov 15 00:19:03 UTC 2023 f81ff285a72e:/opt/STAR-2.7.8a/source
+For more details see:
+<https://github.com/alexdobin/STAR>
+<https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf>
+
+To list all parameters, run STAR --help
+```
+
+An example sbatch script
+
+```bash
+#!/bin/bash
+
+#SBATCH --nodes=1
+#SBATCH --time=01:00:00
+#SBATCH --partition=short
+#SBATCH --job-name=star
+#SBATCH --output=star.out
+#SBATCH --error=star.err
+#SBATCH --ntasks=1
+
+module load singularity/3.10.3
+
+cd /work/mygroup
+
+singularity exec -B "/work:/work" /shared/container_repository/star/star_2.7.8a.sif STAR \
+
+--genomeDIR /path/to/genome \
+--readFilesIn R1.fq R2.fq
+```
+
+## Pull an image from a URL
+
+with the command `singularity pull` you can pull an existing container from the repositories listed below:
+
+Docker Hub https://hub.docker.com/
+
+Singularity Hub https://singularityhub.com/
+
```

Comment:
we can include the nvcr.io repos as well. (Nvidia's container repo).

Reply:
https://catalog.ngc.nvidia.com/containers?filters=&orderBy=weightPopularDESC&query=&page=&pageSize=

Reply:
Awesome. I've add this one in.